### PR TITLE
Displayer

### PR DESCRIPTION
**Pull request by msrasheed**
takes command line arguments now. just type ```python displayer.py -h``` for details.
If the video output is jumpy, it's because the text file doesn't have rows for those frames that the video is skipping over. The displayer will only draw bounding boxes on frames with bounding boxes defined for them in the output text file.

---
This change is [Reviewable](https://reviewable.io/reviews/cam2-hb/re-id/13)

### CODE REVIEW

File: displayer.py [hunk 1]
```diff
@@ -1,73 +1,115 @@
...
+        raise ValueError("vid " + vid + " does not exist")
+
+    invid = cv2.VideoCapture(vid)
+    if invid.isOpened() == False:
+        raise RuntimeError("error opening file " + vid)
+
+    if not os.path.exists(outtxt):
+        raise ValueError("outtxt " + outtxt + "does not exist")
+
+    output = np.loadtxt(outtxt, delimiter=delimiter)
```
Comment (Taher-Dohadwala) [inline displayer.py]: Remove Delimiter, it works without it and crashes with it.

File: displayer.py [hunk 2]
```diff
@@ -1,73 +1,115 @@
...
+        decision = input("the video file " + savename + " already exists. Want to overwrite it? [y/n] ").lower()
+        if decision == 'n':
+            savename = input("new file name: ")
+            savename = "".join(savename.split('.')[:-1]) + ".avi"
+        else:
+            break
+
+    vidfps = int(invid.get(cv2.CAP_PROP_FPS) / interval)
+    vidsize = (int(invid.get(cv2.CAP_PROP_FRAME_WIDTH)), int(invid.get(cv2.CAP_PROP_FRAME_HEIGHT)))
+    outvid = cv2.VideoWriter(savename + '.avi',
```
Comment (Taher-Dohadwala) [inline displayer.py]: Weird bug maybe just for me: The video is not created by outvid. So I entered "dest_vid.avi" in the command line and deleted the +".avi" in outvid and it worked

File: displayer.py [hunk 3]
```diff
@@ -1,73 +1,115 @@
...
+    parser.add_argument("-v", "--view", default=False, action='store_true')
+    parser.add_argument("source_vid", metavar='src', type=str,
+                        help="the source video to use")
+    parser.add_argument("ids_txt", metavar='ids', type=str,
+                        help="the text file of format <frame, id, x1, y1, x2, y2>")
+    parser.add_argument("dest_vid", metavar="dest", type=str,
+                        help="the name of the destination video")
+    return parser.parse_args()
+
+def create_vid(savename, vid, outtxt, view=False):
```
Comment (nhendy) [inline displayer.py]: generally let's stick to the convention of using. underscored variable names to delimit words in a variable name.
So savename -> save_name, outtxt -> out_txt ...etc

File: displayer.py [hunk 4]
```diff
@@ -1,73 +1,115 @@
...
+        raise RuntimeError("error opening file " + vid)
+
+    if not os.path.exists(outtxt):
+        raise ValueError("outtxt " + outtxt + "does not exist")
+
+    output = np.loadtxt(outtxt, delimiter=delimiter)
+    if output.shape[1] != 6:
+        raise ValueError("The text file should have 6 entries per row. yours has {}".format(output.shape[1]))
+
+    uniqframenums = np.sort(np.unique(output[:, frameind]))
```
Comment (nhendy) [inline displayer.py]: uniqframenums -> unique_frame_nums. Could you choose a name that describes what this number is? I think you mean the index. If so stick to using index throughout the code, flipping between frame_num and frame_idx can be confusing.

File: run_reid_pipeline.py [hunk 5]
```diff
@@ -1,297 +1,297 @@
...
-    convert_files_to_numpy(temp_dir, output_files)
-
-    # Run reid model and map track ids to reid ids
-    run_reid_model_and_assign_ids(sort_trackers, attribute_extractor,
-                                  output_files, gallery_feature_vectors)
-
-
-if __name__ == "__main__":
-    sys.exit(main())
+# TODO:(nhendy) script level docstring
```
Comment (nhendy) [inline run_reid_pipeline.py]: Rebase

File: displayer.py [hunk 6]
```diff
@@ -1,73 +1,115 @@
...
+
+# TODO: (nhendy) this script needs massive clean up
+
+frameind = 0
+idind = 1
+x1ind = 2
+y1ind = 3
+x2ind = 4
+y2ind = 5
+delimiter = ','
```
Comment (nhendy) [inline displayer.py]: If we're gonna use static constants can we make them upper case to indicate they're constants?

File: displayer.py [hunk 7]
```diff
@@ -1,73 +1,115 @@
...
+    parser.add_argument("source_vid", metavar='src', type=str,
+                        help="the source video to use")
+    parser.add_argument("ids_txt", metavar='ids', type=str,
+                        help="the text file of format <frame, id, x1, y1, x2, y2>")
+    parser.add_argument("dest_vid", metavar="dest", type=str,
+                        help="the name of the destination video")
+    return parser.parse_args()
+
+def create_vid(savename, vid, outtxt, view=False):
+    """
```
Comment (nhendy) [inline displayer.py]: Another thing: when i saw the variable name `vid` I assumed it was a video object not the path name. Let's make sure the variable names are dumb proof lol . so if it's a path let's call it vide_path ... etc. Usually the way to combat the difficulty of figuring out types in python is by making the variable name as explicit as possible.

**Comment by msrasheed** [comment]
command used to run file
```python displayer.py -d , ../reid-data/msee2/videos/NE_Moiz.mp4 ../reid-data/msee2/NE_Moiz.txt displayer_test.mp4```

**Comment by msrasheed** [comment]
> The final output video size is 500MB which is huge for only 1 min of video.

Output video codec was changed to mp4 from avi so is smaller now

**Comment by msrasheed** [comment]
> Could you add command you used to test to the PR description please? with the expected output.

command used to run file: 
```python displayer.py -d , ../reid-data/msee2/videos/NE_Moiz.mp4 ../reid-data/msee2/NE_Moiz.txt displayer_test.mp4```

Expected output: 
an mp4 file named displayer_test.mp4 with bounding boxes and IDs and stuff drawn on them

**Pull request merged**